src/super_gradients/common/environment/cfg_utils.py

```diff
@@ -57,13 +58,51 @@ def add_params_to_cfg(cfg: DictConfig, params: List[str]):
         cfg.merge_with(new_cfg)
 
 
-def load_arch_params(config_name: str) -> DictConfig:
-    """
-    :param config_name: name of a yaml with arch parameters
-    """
+class ConfigType(Enum):
+    MAIN_RECIPE = "recipe"
+    ANCHORS = "anchors"
+    ARCH_PARAMS = "arch_params"
+    CHECKPOINT_PARAMS = "checkpoint_params"
+    CONVERSION_PARAMS = "conversion_params"
+    DATASET_PARAMS = "dataset_params"
+    QUANTIZATION_PARAMS = "quantization_params"
+    TRAINING_HYPERPARAMS = "training_hyperparams"
+
+
+def load_cfg(config_name, config_type: ConfigType, recipes_dir_path: Optional[str] = None) -> DictConfig:
     GlobalHydra.instance().clear()
-    sg_recipes_dir = pkg_resources.resource_filename("super_gradients.recipes", "")
-    dataset_config = os.path.join("arch_params", config_name)
-    with initialize_config_dir(config_dir=normalize_path(sg_recipes_dir), version_base="1.2"):
+
+    if recipes_dir_path is None:
+        recipes_dir_path = pkg_resources.resource_filename("super_gradients.recipes", "")
```

Comment:
Is this always going to be the case? What if a user places their recipe somewhere else ?

My concern is raised since we are talking about DeciCient users.